### PR TITLE
Move signature type logic to `idb`

### PR DESCRIPTION
**Pull request by tolikzinovyev**
## Summary

Create `SigType` type in `idb/` along with some helper functions.

## Test Plan

Verified that postgres doesn't complain about `SigType` in a query.

### CODE REVIEW

File: idb/sig_type.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	if !stxn.Msig.Blank() {
+		return Msig
+	}
+	if !isZeroSig(stxn.Lsig.Sig) {
+		return Sig
+	}
+	if !stxn.Lsig.Msig.Blank() {
+		return Msig
+	}
+	return Lsig
```
Comment (winder) [inline idb/sig_type.go:41]: It was confusing to see this "default value" when written this way. Could you add an "unknown" type (or error) for the default, and put the 3 lsig options inside a statement like before?

File: idb/sig_type.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+type SigType string
+
+const (
+	Sig  SigType = "sig"
+	Msig SigType = "msig"
+	Lsig SigType = "lsig"
+)
+
+func IsSigTypeValid(sigtype SigType) bool {
+	return (sigtype == Sig) || (sigtype == Msig) || (sigtype == Lsig)
```
Comment (winder) [inline idb/sig_type.go:16]: The lookup map is faster and the SigTypeEnumString means 1 fewer thing to remember when we add a new signature. Could you add it back?

**Review by winder — CHANGES_REQUESTED** [review]
couple small comments otherwise looks good

File: idb/sig_type.go (hunk 1 above)
Comment (tolikzinovyev) [reply idb/sig_type.go]: done

File: idb/sig_type.go (hunk 2 above)
Comment (tolikzinovyev) [reply idb/sig_type.go]: done

**Review by winder — APPROVED** [review]
Thanks for the changes, looks good.

**Pull request merged**